lib/route-using-local-freerouting.ts

```diff
@@ -0,0 +1,163 @@
+        serverReady = true
+      } catch (error) {
+        attempts++
+        await new Promise((resolve) => setTimeout(resolve, 1000))
+      }
+    }
+
+    if (!serverReady) {
+      throw new Error("Server failed to start after multiple attempts")
+    }
+
+    // Create session
+    log("Creating routing session")
+    const sessionResponse = await axios
+      .post(`${API_BASE}/v1/sessions/create`, "", { headers })
+      .catch(handleApiError)
+    const sessionId = sessionResponse.data.id
+    log("Session created with ID: %s", sessionId)
+
+    // Create job
+    log("Creating routing job")
+    const jobResponse = await axios
+      .post(
+        `${API_BASE}/v1/jobs/enqueue`,
+        {
+          session_id: sessionId,
+          name: "circuit-routing",
+          priority: "NORMAL",
+        },
+        { headers },
+      )
+      .catch(handleApiError)
+    const jobId = jobResponse.data.id
+    log("Job created with ID: %s", jobId)
+
+    // Upload DSN file
+    log("Uploading DSN file: %s", inputPath)
+    const fileData = readFileSync(inputPath)
+    await axios.post(
+      `${API_BASE}/v1/jobs/${jobId}/input`,
+      {
+        filename: inputPath,
+        data: Buffer.from(fileData).toString("base64"),
+      },
+      { headers },
+    )
+
+    // Start job
+    log("Starting routing job...")
+    await axios.put(`${API_BASE}/v1/jobs/${jobId}/start`, "", { headers })
+
+    // Wait for completion
+    let isComplete = false
+    attempts = 0
+    const maxAttempts = 20
+
+    while (!isComplete && attempts < maxAttempts) {
+      const jobStatus = await axios.get(`${API_BASE}/v1/jobs/${jobId}`, {
+        headers,
+      })
+
+      log("Job status: %s", jobStatus.data?.state)
+      if (jobStatus.data?.state === "COMPLETED") {
+        isComplete = true
+        break
+      } else if (jobStatus.data?.state === "FAILED") {
+        throw new Error(`Job failed with state: ${jobStatus.data.state}`)
+      } else if (jobStatus.data?.state === "RUNNING") {
+        await new Promise((resolve) => setTimeout(resolve, 3000))
+      } else {
+        await new Promise((resolve) => setTimeout(resolve, 1000))
+      }
+      attempts++
+    }
+
+    if (!isComplete) {
+      throw new Error("Job timed out waiting for completion")
+    }
+
+    // Get output
+    const outputResponse = await axios.get(
+      `${API_BASE}/v1/jobs/${jobId}/output`,
+      { headers },
+    )
+
+    if (!outputResponse.data?.data) {
+      throw new Error("No output received from job")
+    }
+    return Buffer.from(outputResponse.data.data, "base64").toString()
+  } finally {
+    // Cleanup container
+    if (containerId) {
+      try {
+        await execAsync(`docker stop ${containerId}`)
+        await execAsync(`docker rm ${containerId}`)
+      } catch (error: any) {
+        log("Error stopping/removing container: %s", error.message)
+      }
+    }
+    log("Docker container %s stopped and removed", containerId)
+  }
+}
```

Comment:
i would recommend breaking this function up a little bit, it does SOOOOO MUCH right now

Reply:
like starting the docker container and cleaning it up could be an entire function. You could have a class.

Also you might want to use the [dockerode](https://github.com/apocas/dockerode) package which will make this work reliably across systems (I know the existing cli doesn't use it- but it's much better!!!)

I'm not requiring the refactor because i do like how this is relatively well-contained and well-written code